src/core/include/openvino/op/constant.hpp

```diff
@@ -648,6 +648,7 @@ class OPENVINO_API Constant : public Op {
               typename StorageDataType = fundamental_type_for<Type>,
               typename std::enable_if<Type == element::Type_t::string, bool>::type = true>
     void fill_data(const T& value) {
+        fill_data<element::string>(std::string());
```

Comment:
why do we need it if we raise exception below anyway?

Reply:
May be it is better to correct destructor instead of allocating useless memory?

Reply:
Yes there is useless allocation but on exception path and this temporary allocation should not matter so much (any performance is lost due to exception).

To change in `dtor` it would require introduce some state in buffer which has to be set by user (like dummy initialization but less cost) or buffer must be smart enough to change it on initialization internally. The buffer is also dev API and function cannot be used in Constant.hpp.
This is reason why dummy initialization is selected but if that will be an issue it can be optimized in future.